ui/components/component-library/button-base/button-base.js

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import Box from '../../ui/box';
+import { Icon, ICON_NAMES } from '../icon';
+import { Text } from '../text';
+
+import {
+  ALIGN_ITEMS,
+  DISPLAY,
+  JUSTIFY_CONTENT,
+  TEXT_COLORS,
+  TEXT,
+  SIZES,
+  FLEX_DIRECTION,
+} from '../../../helpers/constants/design-system';
+import { BUTTON_SIZES } from './button.constants';
+
+export const ButtonBase = ({
+  as = 'button',
+  block,
+  children,
+  className,
+  size = BUTTON_SIZES.MD,
+  icon,
+  iconPositionRight,
+  loading,
+  disabled,
+  iconProps,
+  ...props
+}) => {
+  return (
+    <Box
+      as={as}
+      paddingLeft={size === BUTTON_SIZES.AUTO ? 0 : 4}
+      paddingRight={size === BUTTON_SIZES.AUTO ? 0 : 4}
+      className={classnames(
+        'mm-button',
+        `mm-button--size-${size}`,
+        {
+          [`mm-button--loading`]: Boolean(loading),
+          [`mm-button--disabled`]: Boolean(disabled),
+          [`mm-button--block`]: Boolean(block),
+        },
```

Comment:
@darkwing pointed out we can simplfy this a bit even if it originally wasn't his intention 😅 
```suggestion
        {
          'mm-button--loading': Boolean(loading),
          'mm-button--disabled': Boolean(disabled),
          'mm-button--block': Boolean(block),
        },
```
I suppose the above types should be all boolean type anyway so not sure if we need `Boolean` or not 🤷